### DISCUSSION TITLE
Add multiple <tr> from multi-level nested components to parent table

### DISCUSSION THREAD

**Discussion by andreskrip** [livewire/livewire#4674] [1. Help] [answered]
Hi, I have a table containing 3-level of component nesting. My task is to build a flat table containing `<tr>`'s returned from nested components.

Each nested components has a similar structure:
```
@foreach(...)
    <tr></tr>
    <tr></tr>
    <tr></tr>
@endforeach
<livewire:nested-component /> // in 3rd level nesting only @foreach
```

As a result, the table should look like:
```
<table>
    <thead></thead>
    <tbody>
    <tr>parent component data</tr>
    <tr>1st level nesting</tr>
    <tr>2nd level nesting</tr>
    <tr>3rd level nesting</tr>
    </tbody>
</table>
```

But the problem - root tag, which I need to wrap my `<tr>` in.

I tried using `<tbody>` as a root tag, but it didn't work because one `<tbody>` is inside another `<tbody>`.

I also tried to make `<tables>` in `<tables>`, but in this case the styling suffers very much

Can you give me any advice?

**Comment by joshhanley** [comment]
@andreskrip sadly I don't think you are going to be able to structure it in that way as Livewire requires a single root element to work properly.

Why are you outputting them in nested components? Can you just put them all in the main component?

If you can provide more details about the context and what the nest components do, maybe we can come up with a better way to structure it all.

Hope this helps!

**Reply by andreskrip** [reply]
@joshhanley I have a large table in a database where data is stored in the following hierarchy:
```
Service
∟Partner
  ∟Landing
    ∟Source
```
![Снимок экрана от 2022-02-16 10-55-31](https://user-images.githubusercontent.com/52978999/154220462-18c5d2a8-e536-4df1-b754-9ec8e1481424.png)

(the screenshot shows an example when one line is returned from a nested components, but in fact there are multiple)

In the parent component, I get data filtered by service. 
When I click on the service, the second-level component is called, and I get data filtered by partner and service, etc.

I use nested components to follow the hierarchy, separate the logic of getting data and additional functionality such as blocking (used for landings and sources).

I tried to follow the hierarchy in the table by putting the child table in the parent 3 times , but it was bad. 

So now I want to make a flat table, and show hierarchy using css.

**Comment by joshhanley** [comment, marked as answer]
@andreskrip right I understand, yeah I don't think nested components are a good idea for that.

I would just put it all into the one main component.

It will mean there is a lot of logic in that component, but you could make use of traits to split that logic up, to make it more easily readable.

**Reply by andreskrip** [reply]
@joshhanley thanks for advice!

**Reply by joshhanley** [reply]
@andreskrip no worries! 🙂
